utils/swift_build_support/swift_build_support/products/product.py

```diff
@@ -291,6 +291,71 @@ def generate_darwin_toolchain_file(self, platform, arch):
 
         return toolchain_file
 
+    def get_linux_abi(self, arch):
+        # Map tuples of (platform, arch) to ABI
+        #
+        # E.x.: Hard ABI or Soft ABI for Linux map to gnueabihf
+        arch_platform_to_abi = {
+            # For now always map to hard float ABI.
+            'armv7': ('arm', 'gnueabihf')
+        }
+
+        # Default is just arch, gnu
+        sysroot_arch, abi = arch_platform_to_abi.get(arch, (arch, 'gnu'))
+        return sysroot_arch, abi
+
+    def get_linux_sysroot(self, platform, arch):
+        if not self.is_cross_compile_target('{}-{}'.format(platform, arch)):
+            return None
+        sysroot_arch, abi = self.get_linux_abi(arch)
+        # $ARCH-$PLATFORM-$ABI
+        # E.x.: aarch64-linux-gnu
+        sysroot_dirname = '{}-{}-{}'.format(sysroot_arch, platform, abi)
+        return os.path.join(os.sep, 'usr', sysroot_dirname)
+
+    def get_linux_target(self, platform, arch):
+        sysroot_arch, abi = self.get_linux_abi(arch)
+        return '{}-unknown-linux-{}'.format(sysroot_arch, abi)
+
+    def generate_linux_toolchain_file(self, platform, arch):
+        shell.makedirs(self.build_dir)
+        toolchain_file = os.path.join(self.build_dir, 'BuildScriptToolchain.cmake')
+
+        toolchain_args = {}
+
+        toolchain_args['CMAKE_SYSTEM_NAME'] = 'Linux'
+        toolchain_args['CMAKE_SYSTEM_PROCESSOR'] = arch
+
+        # We only set the actual sysroot if we are actually cross
+        # compiling. This is important since otherwise cmake seems to change the
+        # RUNPATH to be a relative rather than an absolute path, breaking
+        # certain cmark tests (and maybe others).
+        maybe_sysroot = self.get_linux_sysroot(platform, arch)
+        if maybe_sysroot is not None:
+            toolchain_args['CMAKE_SYSROOT'] = maybe_sysroot
```

Comment:
@drexin @edymtt @compnerd This was an interesting thing that I didn't expect. If I set sys root when cross compiling for the host on the host and set sys root to /, cmake strips the '/' off the front of executables run paths causing dynamic library lookup to failure.